gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-debug/src/main/java/io/gravitee/gateway/jupiter/debug/invoker/DebugInvokerHook.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.jupiter.debug.invoker;
+
+import io.gravitee.gateway.debug.core.invoker.InvokerResponse;
+import io.gravitee.gateway.jupiter.api.ExecutionFailure;
+import io.gravitee.gateway.jupiter.api.ExecutionPhase;
+import io.gravitee.gateway.jupiter.api.hook.InvokerHook;
+import io.gravitee.gateway.jupiter.debug.hook.AbstractDebugHook;
+import io.gravitee.gateway.jupiter.debug.reactor.context.DebugRequestExecutionContext;
+import io.reactivex.Completable;
+import io.reactivex.annotations.Nullable;
+
+/**
+ * @author Guillaume LAMIRAND (guillaume.lamirand at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class DebugInvokerHook extends AbstractDebugHook implements InvokerHook {
+
+    @Override
+    public String id() {
+        return "hook-debug-invoker";
+    }
+
+    @Override
+    protected Completable pre(final String id, final DebugRequestExecutionContext ctx, final ExecutionPhase executionPhase) {
+        return Completable.complete();
+    }
+
+    @Override
+    protected Completable post(
+        final String id,
+        final DebugRequestExecutionContext debugCtx,
+        @Nullable final ExecutionPhase executionPhase
+    ) {
+        return debugCtx
+            .response()
+            .bodyOrEmpty()
+            .doOnSuccess(
+                outputBody -> {
+                    InvokerResponse invokerResponse = debugCtx.getInvokerResponse();
+                    invokerResponse.setHeaders(debugCtx.response().headers());
+                    invokerResponse.setStatus(debugCtx.response().status());
+                    invokerResponse.getBuffer().appendBuffer(outputBody);
+                }
+            )
+            .ignoreElement();
+    }
+
+    @Override
+    protected Completable error(
+        final String id,
+        final DebugRequestExecutionContext debugCtx,
+        final ExecutionPhase executionPhase,
+        final Throwable throwable
+    ) {
+        // FIXME Shouldn't we do something with the throwable?
+        return post(id, debugCtx, executionPhase);
```

Comment:
could be,it is just that there is a throwable that is not used in this case for invoker.

Reply:
Yes I think it could be interesting to handle the error here cause, currently, if an error occurs during backend invocation, we don't see it in the debug.

Maybe it is also the occasion to implement the `interrupt` and `interruptWith` ?

Reply:
actually there is no section about the invoker part so what you are asking for is a new feature.